pypika/tests/test_selects.py

```diff
@@ -378,6 +379,141 @@ def test_where_field_equals_for_update(self):
+                'FOR UPDATE OF {quote_char}abc{quote_char}'.format(
+                    quote_char=quote_char,
+                ),
+                str(q),
+            )
+
+    def test_where_field_equals_for_update_of_multiple_tables(self):
+        for query_cls in [
+            MySQLQuery,
+            PostgreSQLQuery,
+        ]:
+            q = (
+                query_cls.from_(self.t)
+                .join(self.t2)
+                .on(self.t.id == self.t2.abc_id)
+                .select("*")
+                .where(self.t.foo == self.t.bar)
+                .for_update(of=("abc", "cba"))
+            )
+            quote_char = query_cls._builder().QUOTE_CHAR if isinstance(query_cls._builder().QUOTE_CHAR, str) else '"'
+            self.assertIn(
+                str(q),
+                [
+                    'SELECT * '
+                    'FROM {quote_char}abc{quote_char} '
+                    'JOIN {quote_char}cba{quote_char} '
+                    'ON {quote_char}abc{quote_char}.{quote_char}id{quote_char}='
+                    '{quote_char}cba{quote_char}.{quote_char}abc_id{quote_char} '
+                    'WHERE {quote_char}abc{quote_char}.{quote_char}foo{quote_char}='
+                    '{quote_char}abc{quote_char}.{quote_char}bar{quote_char} '
+                    'FOR UPDATE OF {quote_char}cba{quote_char}, {quote_char}abc{quote_char}'.format(
+                        quote_char=quote_char,
+                    ),
+                    'SELECT * '
+                    'FROM {quote_char}abc{quote_char} '
+                    'JOIN {quote_char}cba{quote_char} '
+                    'ON {quote_char}abc{quote_char}.{quote_char}id{quote_char}='
+                    '{quote_char}cba{quote_char}.{quote_char}abc_id{quote_char} '
+                    'WHERE {quote_char}abc{quote_char}.{quote_char}foo{quote_char}='
+                    '{quote_char}abc{quote_char}.{quote_char}bar{quote_char} '
+                    'FOR UPDATE OF {quote_char}abc{quote_char}, {quote_char}cba{quote_char}'.format(
+                        quote_char=quote_char,
+                    ),
+                ],
+            )
+
+    def test_where_field_equals_for_update_all(self):
+        for query_cls in [
+            MySQLQuery,
+            PostgreSQLQuery,
+        ]:
+            q = (
+                query_cls.from_(self.t)
+                .select("*")
+                .where(self.t.foo == self.t.bar)
+                .for_update(nowait=True, skip_locked=True, of=("abc",))
+            )
+            quote_char = query_cls._builder().QUOTE_CHAR if isinstance(query_cls._builder().QUOTE_CHAR, str) else '"'
+            self.assertEqual(
+                'SELECT * '
+                'FROM {quote_char}abc{quote_char} '
+                'WHERE {quote_char}foo{quote_char}={quote_char}bar{quote_char} '
+                'FOR UPDATE OF {quote_char}abc{quote_char} NOWAIT'.format(
```

Comment:
So SKIP LOCKED shouldn't be here?

Reply:
The name suggests it should be both, but query synopsis for both PostgreSQL and MySQL says that `NOWAIT` and `SKIP LOCKED` are mutually exclusive.
PostgreSQL:
```
[ FOR { UPDATE | NO KEY UPDATE | SHARE | KEY SHARE } [ OF table_name [, ...] ] [ NOWAIT | SKIP LOCKED ] [...] ]
```
MySQL:
```
[FOR {UPDATE | SHARE}
        [OF tbl_name [, tbl_name] ...]
        [NOWAIT | SKIP LOCKED]
      | LOCK IN SHARE MODE]
```

So I split the test into two - testing both cases.

Reply:
I see. Thanks.